examples/src/threejs/post-processing-effects.ts

```diff
@@ -8,9 +8,111 @@ import { CadNode } from '@cognite/reveal/threejs';
 import CameraControls from 'camera-controls';
 
 const postprocessing = require('postprocessing');
+const {
+  BloomEffect,
+  EffectComposer,
+  NormalPass,
+  DepthEffect,
+  RenderPass,
+  BlendFunction,
+  EffectPass,
+  Pass,
+  SSAOEffect,
+  Resizer
+} = postprocessing;
 
 CameraControls.install({ THREE });
 
+function traverseShaderMaterials(rootNode: THREE.Object3D, callback: (material: THREE.ShaderMaterial) => void) {
+  rootNode.traverseVisible(obj => {
+    // NOTE we cannot rely on instanceof here because THREE is not necessarily the same library
+    if (obj.type !== 'Mesh') {
+      return;
+    }
+    const mesh = obj as THREE.Mesh;
+    if (mesh.material == null || Array.isArray(mesh.material) || mesh.material.type !== 'ShaderMaterial') {
+      return;
+    }
+    const material = mesh.material as THREE.ShaderMaterial;
+
+    callback(material);
+  });
+}
+
+class RevealNormalPass extends Pass {
+  renderTarget: THREE.WebGLRenderTarget;
+  scene: THREE.Scene;
+  camera: THREE.Camera;
+  renderToScreen: boolean;
+  private _renderPass: any; // RenderPass
+
+  constructor(scene: THREE.Scene, camera: THREE.Camera) {
+    super('RevealNormalPass');
+
+    // @ts-ignore
```

Comment:
Why? Can't this just be declared? Also, stick to either member vars with/without "_"-prefix